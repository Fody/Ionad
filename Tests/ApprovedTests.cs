using System.Threading.Tasks;
using Fody;
using VerifyTests;
using VerifyXunit;
using Xunit;

public partial class ModuleWeaverTests
{
    [Fact]
    public Task ClassWithBrokenReplacement()
    {
        var settings = new VerifySettings();
        settings.UniqueForRuntime();
        return Verifier.Verify(Ildasm.Decompile(testResult.AssemblyPath, "ClassWithBrokenReplacement"), settings);
    }

    [Fact]
    public Task ClassWithDateTime()
    {
        var settings = new VerifySettings();
        settings.UniqueForRuntime();
        return Verifier.Verify(Ildasm.Decompile(testResult.AssemblyPath, "ClassWithDateTime"), settings);
    }

    [Fact]
    public Task ClassWithGenericMethodUsage()
    {
        var settings = new VerifySettings();
        settings.UniqueForRuntime();
        return Verifier.Verify(Ildasm.Decompile(testResult.AssemblyPath, "ClassWithGenericMethodUsage"), settings);
    }

    [Fact]
    public Task ClassWithGenericUsage()
    {
        var settings = new VerifySettings();
        settings.UniqueForRuntime();
        return Verifier.Verify(Ildasm.Decompile(testResult.AssemblyPath, "ClassWithGenericUsage"), settings);
    }
}