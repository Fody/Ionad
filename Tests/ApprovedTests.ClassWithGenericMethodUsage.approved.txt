
//  Copyright (c) Microsoft Corporation.  All rights reserved.


// warning : THIS IS A PARTIAL DISASSEMBLY, NOT SUITABLE FOR RE-ASSEMBLING



.class public auto ansi beforefieldinit ClassWithGenericMethodUsage
       extends [mscorlib]System.Object
{
  .method public hidebysig instance void 
          Method() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  1
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    .line 4,4 : 5,6 'AssemblyToProcess\\ClassWithGenericMethodUsage.cs'
    IL_0000:  nop
    .line 5,5 : 9,50 ''
    IL_0001:  call       !!0 StaticWithGenericMethodReplacement::Method<string>()
    IL_0006:  pop
    .line 6,6 : 5,6 ''
    IL_0007:  ret
  } // end of method ClassWithGenericMethodUsage::Method

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } // end of method ClassWithGenericMethodUsage::.ctor

} // end of class ClassWithGenericMethodUsage

// *********** DISASSEMBLY COMPLETE ***********************
